WalletWasabi.Tests/UnitTests/Hwi/MockedDeviceTests.cs

```diff
@@ -0,0 +1,372 @@
+				Assert.False(entry.IsInitialized());
+				Assert.Null(entry.Fingerprint);
+
+				var deviceType = entry.Model;
+				var devicePath = entry.Path;
+
+				await client.WipeAsync(deviceType, devicePath, cts.Token);
+				await client.SetupAsync(deviceType, devicePath, false, cts.Token);
+				await client.RestoreAsync(deviceType, devicePath, false, cts.Token);
+
+				// Trezor T doesn't support it.
+				var promptpin = await Assert.ThrowsAsync<HwiException>(async () => await client.PromptPinAsync(deviceType, devicePath, cts.Token));
+				Assert.Equal("The PIN has already been sent to this device", promptpin.Message);
+				Assert.Equal(HwiErrorCode.DeviceAlreadyUnlocked, promptpin.ErrorCode);
+
+				var sendpin = await Assert.ThrowsAsync<HwiException>(async () => await client.SendPinAsync(deviceType, devicePath, 1111, cts.Token));
+				Assert.Equal("The PIN has already been sent to this device", sendpin.Message);
+				Assert.Equal(HwiErrorCode.DeviceAlreadyUnlocked, sendpin.ErrorCode);
+
+				KeyPath keyPath1 = KeyManager.DefaultAccountKeyPath;
+				KeyPath keyPath2 = KeyManager.DefaultAccountKeyPath.Derive(1);
+				ExtPubKey xpub1 = await client.GetXpubAsync(deviceType, devicePath, keyPath1, cts.Token);
+				ExtPubKey xpub2 = await client.GetXpubAsync(deviceType, devicePath, keyPath2, cts.Token);
+				var expecteXpub1 = NBitcoinHelpers.BetterParseExtPubKey("xpub6DHjDx4gzLV37gJWMxYJAqyKRGN46MT61RHVizdU62cbVUYu9L95cXKzX62yJ2hPbN11EeprS8sSn8kj47skQBrmycCMzFEYBQSntVKFQ5M");
+				var expecteXpub2 = NBitcoinHelpers.BetterParseExtPubKey("xpub6FJS1ne3STcKdQ9JLXNzZXidmCNZ9dxLiy7WVvsRkcmxjJsrDKJKEAXq4MGyEBM3vHEw2buqXezfNK5SNBrkwK7Fxjz1TW6xzRr2pUyMWFu");
+				Assert.Equal(expecteXpub1, xpub1);
+				Assert.Equal(expecteXpub2, xpub2);
+
+				BitcoinWitPubKeyAddress address1 = await client.DisplayAddressAsync(deviceType, devicePath, keyPath1, cts.Token);
+				BitcoinWitPubKeyAddress address2 = await client.DisplayAddressAsync(deviceType, devicePath, keyPath2, cts.Token);
+
+				BitcoinAddress expectedAddress1;
+				BitcoinAddress expectedAddress2;
+				if (network == Network.Main)
+				{
+					expectedAddress1 = BitcoinAddress.Create("bc1q7zqqsmqx5ymhd7qn73lm96w5yqdkrmx7fdevah", Network.Main);
+					expectedAddress2 = BitcoinAddress.Create("bc1qmaveee425a5xjkjcv7m6d4gth45jvtnj23fzyf", Network.Main);
+				}
+				else if (network == Network.TestNet)
+				{
+					expectedAddress1 = BitcoinAddress.Create("tb1q7zqqsmqx5ymhd7qn73lm96w5yqdkrmx7rtzlxy", Network.TestNet);
+					expectedAddress2 = BitcoinAddress.Create("tb1qmaveee425a5xjkjcv7m6d4gth45jvtnjqhj3l6", Network.TestNet);
+				}
+				else if (network == Network.RegTest)
+				{
+					expectedAddress1 = BitcoinAddress.Create("bcrt1q7zqqsmqx5ymhd7qn73lm96w5yqdkrmx7pzmj3d", Network.RegTest);
+					expectedAddress2 = BitcoinAddress.Create("bcrt1qmaveee425a5xjkjcv7m6d4gth45jvtnjz7tugn", Network.RegTest);
+				}
+				else
+				{
+					throw new NotSupportedException($"{network} not supported.");
```

Comment:
```suggestion
					throw new NotSupportedNetworkException(network);
```
Don't forget to add `using WalletWasabi.Exceptions;`

Reply:
Done.